tasks/shared_libraries/recovery/test/unit_tests/test_shared_recovery.py

```diff
@@ -106,149 +245,225 @@ def test_post_status_for_file_to_queue_RequestPost_StatusSuccess(self):
-#             job_id= '1234'
-#             granule_id= '6c8d0c8b-4f9a-4d87-ab7c-480b185a0250'
-#             status_id = shared_recovery.OrcaStatus.SUCCESS
-#             request_method = shared_recovery.RequestMethod.NEW
-#             archive_destination = 'archive-destination'
-#             request_time = '2019-07-17T17:36:38.494918'
-#             completion_time = '2019-07-18T17:36:38.494918'
-#             db_queue_url = self.queue.url
-#             #this is the expected message body that should be received
-#             new_data = {"job_id": job_id, "granule_id": granule_id, "request_time": request_time, 
-#                         "completion_time": completion_time, "archive_destination": archive_destination }
-#             body = json.dumps(new_data)
-#             MessageDeduplicationId = table_name + request_method.value + body
-#             MessageAttributes={
-#                 "RequestMethod": {
-#                     "DataType": "String",
-#                     "StringValue": request_method.value,
-#                 },
-#                 "TableName": {"DataType": "String", "StringValue": table_name},
-#             }
-#             shared_recovery.post_status_for_job_to_queue(
-#                 job_id,granule_id,status_id,
-#                 request_time,completion_time,archive_destination,request_method,db_queue_url
-#                 )
-#             #grabbing queue contents after the message is sent
-#             queue_contents = self.queue.receive_messages(
-#                             MessageAttributeNames= ["All"]
-#                             )
-
-#             self.assertEqual(queue_contents[0].body, json.dumps(new_data))
-#             self.assertEqual(queue_contents[0].attributes['MessageDeduplicationId'], MessageDeduplicationId)
-#             self.assertEqual(queue_contents[0].attributes['MessageGroupId'], self.MessageGroupId)
-#             self.assertEqual(queue_contents[0].message_attributes, MessageAttributes)
+    def test_post_status_for_job_to_queue_new_pending(self):
+            """
+            Test that sending a message to SQS queue using post_status_for_job_to_queue() function returns the same expected message.
+            This is for testing the case when request_method is NEW, ORCA status is PENDING.
+            The new_data in this case should return request_time, archive_destination in addition to other values.
+            Completion_time should not exist in the response received from SQS body.
+            """
+            table_name = 'orca_recoveryjob'
+            request_method = shared_recovery.RequestMethod.NEW
+            job_id= '1234'
+            granule_id= '6c8d0c8b-4f9a-4d87-ab7c-480b185a0250'
+            status_id = shared_recovery.OrcaStatus.PENDING
+            request_method = shared_recovery.RequestMethod.NEW
+            archive_destination = 'archive-bucket'
+            request_time = datetime.now(timezone.utc).isoformat()
```

Comment:
request_time and completion_time are created are generated and do not need to be set since you are not mocking out those variables. 

- test_post_status_for_job_to_queue_new_pending
- test_post_status_for_job_to_queue_new_staged
- test_post_status_for_job_to_queue_new_success
- test_post_status_for_job_to_queue_new_failed

Consider parameterize the tests. See previous comment.

I noticed that you never set the `request_method` value to `shared_recovery.RequestMethod.UPDATE` in any of your tests. Isn't that a valid value path that should be tested?

Reply:
removed request_time, last_update, or completion_time.
Moved some of the repetitive parameters under setUp function to minimize repetition.